File: Day20/src/test/java/uk/co/stevebosman/aoc24/day20/SolverTest.java
package uk.co.stevebosman.aoc24.day20;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertEquals;

class SolverTest {
  @ParameterizedTest
  @CsvSource({
          "2, 44",
          "4, 30",
          "6, 16",
          "8, 14",
          "10, 10",
          "12, 8",
          "20, 5",
          "36, 4",
          "38, 3",
          "40, 2",
          "64, 1",
  })
  void example1Part1(final long target, final long expected) throws IOException {
    final long actual = assertDoesNotThrow(() -> new Solver("example1.txt")).cheatsSaveAtLeast(target, 2);
    assertEquals(expected, actual);
  }

  @ParameterizedTest
  @CsvSource({
          "74, 7",
          "76, 3",
  })
  void example1Part2(final long target, final long expected) throws IOException {
    final long actual = assertDoesNotThrow(() -> new Solver("example1.txt")).cheatsSaveAtLeast(target, 20);
    assertEquals(expected, actual);
  }

  @Test
  void inputPart1() throws IOException {
    final long actual = assertDoesNotThrow(() -> new Solver("input.txt")).cheatsSaveAtLeast(100, 2);
    System.out.println("Part1 = " + actual);
  }

  @Test
  void inputPart2() throws IOException {
    final long actual = assertDoesNotThrow(() -> new Solver("input.txt")).cheatsSaveAtLeast(100, 20);
    System.out.println("Part2 = " + actual);
  }
}

File: Day07/src/test/kotlin/SolverTest.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.CsvSource
import uk.co.stevebosman.example.Solver

class SolverTest {

    @ParameterizedTest
    @CsvSource(
        "190, 10 19, true",
        "3267, 81 40 27, true",
        "83, 17 5, false",
        "156, 15 6, false",
        "7290, 6 8 6 15, false",
        "161011, 16 10 13, false",
        "192, 17 8 14, false",
        "21037, 9 7 18 13, false",
        "292, 11 6 16 20, true"
    )
    fun isSolvable(target: Long, values: String, expected: Boolean) {
        assertEquals(expected, Solver().isSolvable(target, values.split(" ").map { s -> s.toLong() }))
    }

}